images/build/Dockerfiles/vso.bullseye.Dockerfile

```diff
@@ -0,0 +1,325 @@
+        tk-dev \
+        uuid-dev \
+        curl \
+        gnupg \
+    && rm -rf /var/lib/apt/lists/* \
+    # Install .NET Core SDKs
+    && nugetPackagesDir="/var/nuget" \
+    && mkdir -p $nugetPackagesDir \
+    # Grant read-write permissions to the nuget folder so that dotnet restore
+    # can write into it.
+    && chmod a+rw $nugetPackagesDir \
+    && DOTNET_RUNNING_IN_CONTAINER=true \
+    && DOTNET_USE_POLLING_FILE_WATCHER=true \
+	&& NUGET_XMLDOC_MODE=skip \
+    && DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1 \
+	&& NUGET_PACKAGES="$nugetPackagesDir" \
+    && . $buildDir/__dotNetCoreSdkVersions.sh \
+    && DOTNET_SDK_VER=$DOT_NET_CORE_31_SDK_VERSION \
+       INSTALL_PACKAGES="true" \
+       $imagesDir/build/installDotNetCore.sh \
+    && DOTNET_SDK_VER=$DOT_NET_60_SDK_VERSION \
+       INSTALL_PACKAGES="true" \
+       $imagesDir/build/installDotNetCore.sh \
+    && rm -rf /tmp/NuGetScratch \
+    && find $nugetPackagesDir -type d -exec chmod 777 {} \; \
+    && cd /opt/dotnet \
+    && . $buildDir/__dotNetCoreSdkVersions.sh \
+    && ln -s $DOT_NET_CORE_31_SDK_VERSION 3-lts \
+    && ln -s $DOT_NET_60_SDK_VERSION 6-lts \
+    && ln -s $DOT_NET_60_SDK_VERSION lts \
+    # Install Hugo
+    && mkdir -p /home/codespace/.hugo \
+    && $imagesDir/build/installHugo.sh \
+    # Install Node
+    && mkdir -p /home/codespace/.nodejs \
+    && . $buildDir/__nodeVersions.sh \
+    && $imagesDir/installPlatform.sh nodejs $NODE14_VERSION \
+    && $imagesDir/installPlatform.sh nodejs $NODE16_VERSION \
+    && $imagesDir/receiveGpgKeys.sh 6A010C5166006599AA17F08146C2130DFD2497F5 \
+    && curl -fsSLO --compressed "https://yarnpkg.com/downloads/$YARN_VERSION/yarn-v$YARN_VERSION.tar.gz" \
+    && curl -fsSLO --compressed "https://yarnpkg.com/downloads/$YARN_VERSION/yarn-v$YARN_VERSION.tar.gz.asc" \
+    && gpg --batch --verify yarn-v$YARN_VERSION.tar.gz.asc yarn-v$YARN_VERSION.tar.gz \
+    && mkdir -p /opt/yarn \
+    && tar -xzf yarn-v$YARN_VERSION.tar.gz -C /opt/yarn \
+    && mv /opt/yarn/yarn-v$YARN_VERSION /opt/yarn/$YARN_VERSION \
+    && rm yarn-v$YARN_VERSION.tar.gz.asc yarn-v$YARN_VERSION.tar.gz \
+    && cd /opt/nodejs \
+    && ln -s $NODE14_VERSION 14 \
+    && ln -s $NODE16_VERSION 16 \
+    && ln -s $NODE16_VERSION lts \
+    && ln -sfn /opt/nodejs/$NODE16_VERSION /home/codespace/.nodejs/current \
+    && cd /opt/yarn \
+    && ln -s $YARN_VERSION stable \
+    && ln -s $YARN_VERSION latest \
+    && ln -s $YARN_VERSION $YARN_MINOR_VERSION \
+    && ln -s $YARN_MINOR_VERSION $YARN_MAJOR_VERSION \
+    # Install Python SDKs
+    # Upgrade system python
+    && mkdir -p /home/codespace/.python \
+    && pip install --upgrade cython \
+    && pip3 install --upgrade cython \
+    && . $buildDir/__pythonVersions.sh \
+    && $imagesDir/installPlatform.sh python $PYTHON39_VERSION \
+    # && $imagesDir/installPlatform.sh python $PYTHON310_VERSION \
+    # build python directly from source
```

Comment:
nit: just curious why we would want to build Python 3.10 from source here rather than use the `installPlatform.sh` script to do this

Reply:
Originally using `installPlatform.sh` for python 3.10 was failing, but I figured out why and have updated this. 

The reason this was failing was because the Debain flavor was still set as focal, which caused focal SDK to be downloaded, which is incompatible with this bullseye image.